pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -240,13 +246,28 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 						cr.Name,
 					)
 					updateMetaGenerator := &kubecluster
-					updateMetaGenerator.Spec.ConnectionsGenToProcess = kubecluster.Spec.ConnectionsGenToProcess + 1
+					//updateMetaGenerator.Spec.ConnectionsGenToProcess = kubecluster.Spec.ConnectionsGenToProcess + 1
+					annotations := updateMetaGenerator.Annotations
+					if v, ok := annotations[connectionsIncrementor]; ok {
+						newV, _ := strconv.Atoi(v)
+						annotations[connectionsIncrementor] = strconv.Itoa(newV + 1)
+					} else {
+						annotations[connectionsIncrementor] = "1"
+					}
+					updateMetaGenerator.Annotations = annotations
```

Comment:
Seems unnecessary to build the annotations here if we're just going to do it again in the top of the loop below...